File: SuperMovie_server/src/api/Schedule/DeleteSchedule.cs
namespace SuperMovie.Server.Api.Schedule;

using WebSocketSharp;
using WebSocketSharp.Server;

public struct DeleteScheduleReq
{
    public long ScheduleId;
}

public struct DeleteScheduleRsp
{
    public bool Ok;
}

//api : delete_schedule
public class DeleteSchedule : WebSocketBehavior
{
}

File: movieui/src/scripts/ws/Schedule/deleteSchedule.ts
import {wsUrlRoot} from "@/scripts/ws/meta"
import {recvMsg, reqStringify, rspParse, sendMsg} from "@/scripts/ws/helper"

export {deleteSchedule}

const conn = new WebSocket(`${wsUrlRoot}/delete_schedule`)

type DeleteScheduleReq =
    {
        ScheduleId: bigint
    }

type DeleteScheduleRsp =
    {
        Ok: boolean
    }

async function deleteSchedule(req: DeleteScheduleReq) {

    const task = recvMsg(conn)
    sendMsg(conn, reqStringify(req)).then()
    const msg = await task

    return <DeleteScheduleRsp>rspParse(msg)
}